### PR TITLE
BREAKING: Upgrade maplibre-gl to v2

### PR DESCRIPTION
**Pull request by slaymance**
#### Description of changes
The [maplibre-gl v2 release](https://github.com/maplibre/maplibre-gl-js/releases/tag/v2.0.0) included several breaking changes, mostly regarding TypeScript type exports. This PR bumps the `peerDependency` and `devDependency` on `maplibre-gl` to the latest version and fixes the build-time, run-time, and testing errors generated as a result of the library upgrade.

#### Issue #, if available
#107 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added]
- [x] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

**Comment by thaddmt** [comment]
Looks awesome so far @slaymance !

File: __tests__/AmplifyGeofenceControl.test.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ describe("AmplifyGeofenceControl", () => {
     const control = new AmplifyGeofenceControl({
       geofenceCollectionId: "anyString",
     });
-    control._map = new maplibreMap();
+    control._map = new maplibreMap({ container: '', style: '' });
```
Comment (thaddmt) [inline __tests__/AmplifyGeofenceControl.test.ts:75]: @slaymance is this the required arguments you were referring to during standup?

Comment (slaymance) [reply __tests__/AmplifyGeofenceControl.test.ts:75]: Yep, the `MapOptions` type in v2 has those [properties marked as required](https://github.com/maplibre/maplibre-gl-js/blob/1f123972d2e80498e437cf63be24ae33a059888c/src/ui/map.ts#L62-#L107), while in v1 it accepted `MapboxOptions` which [had no such requirements](https://github.com/maplibre/maplibre-gl-js/blob/v1.15.2/src/index.d.ts#L235).

Comment (thaddmt) [reply __tests__/AmplifyGeofenceControl.test.ts:75]: Awesome yep this change should be fine as effectively there will be no change to the `createMap` function right?

**Comment by thaddmt** [comment]
I believe if we prefix this PR with `BREAKING` or `perf` (per semantic release docs) it will do the major version bump for us when we square and merge

**Review by thaddmt — APPROVED** [review]

**Review by thaddmt — APPROVED** [review]

**Pull request merged**